staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go

```diff
@@ -445,12 +445,11 @@ type StorageProfileStatus struct {
 type ClaimPropertySet struct {
 	// AccessModes contains the desired access modes the volume should have.
 	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes-1
-	// +optional
-	AccessModes []corev1.PersistentVolumeAccessMode `json:"accessModes,omitempty" protobuf:"bytes,1,rep,name=accessModes,casttype=PersistentVolumeAccessMode"`
+	AccessModes []corev1.PersistentVolumeAccessMode `json:"accessModes"`
```

Comment:
afaik there's no `kubebuilder:validation` marker for validating enum values for a slice (`AccessModes`) like addded for `VolumeMode`. The only ways I can think of is either by a validating webhook, or by adding an alias type, e.g.:
```go
+// +kubebuilder:validation:Enum="ReadWriteOnce";"ReadOnlyMany";"ReadWriteMany";"ReadWriteOncePod"
+type ClaimPropertySetAccessMode corev1.PersistentVolumeAccessMode
```
Both solutions aren't so nice.

Reply:
I don't mind the second option, creating a type that is just the same, and adding the validation there.